qutebrowser/javascript/webengine_caret.js

```diff
@@ -0,0 +1,1401 @@
+     */
+    CaretBrowsing.setInitialCursor = function(platform) {
+        if (platform) {
+            CaretBrowsing.isWindows = platform.indexOf("Windows") !== -1;
+        }
+        const selectionRange = window.getSelection().toString().length;
+        if (selectionRange === 0) {
+            positionCaret();
+        }
+
+        CaretBrowsing.injectCaretStyles();
+        CaretBrowsing.toggle();
+        CaretBrowsing.initiated = true;
+        CaretBrowsing.selectionEnabled = selectionRange > 0;
+    };
+
+    /**
+     * Try to set the window's selection to be between the given start and end
+     * cursors, and return whether or not it was successful.
+     * @param {Cursor} start The start position.
+     * @param {Cursor} end The end position.
+     * @return {boolean} True if the selection was successfully set.
+     */
+    CaretBrowsing.setAndValidateSelection = function(start, end) {
+        const sel = window.getSelection();
+        sel.setBaseAndExtent(start.node, start.index, end.node, end.index);
+
+        if (sel.rangeCount !== 1) {
+            return false;
+        }
+
+        return (sel.anchorNode === start.node &&
+            sel.anchorOffset === start.index &&
+            sel.focusNode === end.node &&
+            sel.focusOffset === end.index);
+    };
+
+    /**
+     * Set focus to a node if it's focusable. If it's an input element,
+     * select the text, otherwise it doesn't appear focused to the user.
+     * Every other control behaves normally if you just call focus() on it.
+     * @param {Node} node The node to focus.
+     * @return {boolean} True if the node was focused.
+     */
+    CaretBrowsing.setFocusToNode = function(nodeArg) {
+        let node = nodeArg;
+        while (node && node !== document.body) {
+            if (isFocusable(node) && node.constructor !== HTMLIFrameElement) {
+                node.focus();
+                if (node.constructor === HTMLInputElement && node.select) {
+                    node.select();
+                }
+                return true;
+            }
+            node = node.parentNode;
+        }
+
+        return false;
+    };
+
+    /**
+     * Set the caret element's normal style, i.e. not when animating.
+     */
+    CaretBrowsing.setCaretElementNormalStyle = function() {
+        const element = CaretBrowsing.caretElement;
+        element.className = "CaretBrowsing_Caret";
+        if (CaretBrowsing.isSelectionCollapsed) {
+            element.style.opacity = "1.0";
+        } else {
+            element.style.opacity = "0.0";
+        }
+        element.style.left = `${CaretBrowsing.caretX}px`;
+        element.style.top = `${CaretBrowsing.caretY}px`;
+        element.style.width = `${CaretBrowsing.caretWidth}px`;
+        element.style.height = `${CaretBrowsing.caretHeight}px`;
+        element.style.color = CaretBrowsing.caretForeground;
+    };
+
+    /**
+     * Animate the caret element into the normal style.
```

Comment:
Since we turned off animations/flashing, I think this, `flashCaretElement`, the calling code, and stuff related to it in the stylesheet (`CaretBrowsing_AnimateCaret`) could all be removed?